### PR TITLE
Prometheus: Fix exemplar fill color to match series color in time series.

### LINKED ISSUE #59499: Prometheus: Exemplars not color coding
**What happened**:
Exemplars are not color coded, and exemplars for different series are not visually differentiable. 

image

**What you expected to happen**:
Exemplar points for different metrics would match the color of the metric on the graph.

**How to reproduce it (as minimally and precisely as possible)**:
Spin up the TNS app, create graph with multiple series/exemplars. Or view the client dashboard link provided in the support (https://github.com/grafana/support-escalations/issues/4505).

**Anything else we need to know?**:
It appears that each metric gets a different icon, but for queries within a single metric, with many series/labels, can the time series chart use a different style treatment for each series? Could they even match the color of the time series graph itself?

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Matching exemplar color to the color of the series.

**Why do we need this feature?**
Current exemplar UX is difficult to differentiate exemplars from different labels within a single metric.

**Who is this feature for?**
Customers with exemplars that want to build more complicated graphs.

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana/issues/59499

Debug log:
`histogram_quantile(0.90, sum by(le, job, status_code) (rate(tns_request_duration_seconds_bucket[$__rate_interval])))`
[debug-Exemplar 1 metric 10 labels-2022-12-07 16_00_31.json.txt](https://github.com/grafana/grafana/files/10180336/debug-Exemplar.1.metric.10.labels-2022-12-07.16_00_31.json.txt)

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/timeseries/plugins/ExemplarsPlugin.tsx [hunk 1]
```diff
@@ -146,7 +142,7 @@ export const getVisibleLabels = (
         const field = frames[frameIndex].fields[fieldIndex];
         if (field.labels) {
           // Note that this may be an empty object in the case of a metric being rendered with no labels
-          visibleLabels.push(field.labels);
+          visibleLabels.push({ ...field.labels, __color__: plotInstance.props?.lineColor ?? '' });
```
Comment (gtk-grafana) [inline public/app/plugins/panel/timeseries/plugins/ExemplarsPlugin.tsx:145]: maybe it's time to revisit this?

File: public/app/plugins/panel/timeseries/plugins/ExemplarsPlugin.tsx [hunk 2]
```diff
@@ -156,21 +152,51 @@ export const getVisibleLabels = (
 };
 
 /**
- * Determine if the current exemplar marker is filtered by what series are selected in the legend UI
+ * Get color of active series in legend
+ */
+const getExemplarColor = (
```
[the remark this answers is not in the document]
Comment (gtk-grafana) [reply public/app/plugins/panel/timeseries/plugins/ExemplarsPlugin.tsx:162]: Actually this appears to be an issue with the base branch!

[the remark this answers is not in the document]
Comment (gtk-grafana) [reply public/app/plugins/panel/timeseries/plugins/ExemplarsPlugin.tsx:162]: Fixed and merged

**Comment by gtk-grafana** [comment]
image

image

image

image

image

**Review by itsmylife — APPROVED** [review]
LGTM

**Pull request merged**